lib/fs-hash-diff.js

```diff
@@ -0,0 +1,77 @@
+/// @ts-check
+"use strict";
+
+// Imported for type annotations.
+const Entry = require("fs-tree-diff/lib/entry").default; // jshint ignore:line
+const FSTree = require("fs-tree-diff");
+
+class HashEntry {
+  /**
+   * @param relativePath {string}
+   * @param hash {string}
+   */
+  constructor(relativePath, hash) {
+    this.relativePath = relativePath;
+    this.hash = hash;
+  }
+
+  isDirectory() {
+    return false;
+  }
+
+  /**
+   * Whether the entries have the same content.
+   * @param other {unknown}
+   * @returns
+   */
+  equals(other) {
+    if (other instanceof HashEntry) {
+      return this.hash === other.hash;
+    } else {
+      return false;
+    }
+  }
+}
+
+class FSHashTree extends FSTree {
+  /**
+   * Creates an instance of FSHashTree.
+   * @param [options] {{entries?: Array<Entry|HashEntry>, sortAndExpand?: boolean}}
+   */
+  constructor(options) {
+    super(options);
+    /** @type Array<Entry|HashEntry> */
+    this.entries = options.entries || [];
```

Comment:
I believe the base-class has already set `this.entries`, this line is likely un-needed.

In-fact, I suspect we can simply use the inherited constructor.

As it both [sets entries](https://github.com/stefanpenner/fs-tree-diff/blob/0f7eca7d74b34113496dc23d59b45a71b6f61e8a/lib/index.ts#L81) and [default its to](https://github.com/stefanpenner/fs-tree-diff/blob/0f7eca7d74b34113496dc23d59b45a71b6f61e8a/lib/index.ts#L73) `[]` if not set.